File: dfs_bfs/boj/p11724.py
import sys

def dfs(v):
    visited[v] = 1
    for i in range(1, n+1):
        if visited[i] == 0 and graph[v][i] == 1:
            dfs(i)

sys.setrecursionlimit(10000) # 안하면 런타임 오류 재귀 오류 남
n, m = map(int, sys.stdin.readline().split())
graph = [[0] * (n+1) for i in range(n+1)]
visited = [0] * (n+1)

for i in range(m):
    v1, v2 = map(int, sys.stdin.readline().split())
    graph[v1][v2] = 1
    graph[v2][v1] = 1

count = 0
for i in range(1, n+1):
    if visited[i] == 0:
        dfs(i)
        count += 1
print(count)

File: dfs_bfs/boj/P11724.java
package dfs_bfs.boj;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

class P11724 {
    static int n, m;
    static int[][] graph;
    static boolean[]  visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());

        graph = new int[n+1][n+1];

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());

            graph[x][y] = 1;
            graph[y][x] = 1;
        }

        visited = new boolean[n+1];
        int count = 0;

        for (int i = 1; i <= n; i++) {
            if (visited[i] == false) {
                bfs(i);
                count++;
            }
        }
        
        System.out.println(count);
    }

    static void dfs(int node) {
        if (visited[node] == true) {
            return;
        }

        visited[node] = true;

        for (int i = 1; i <=n; i++) {
            if (graph[node][i] == 1 && visited[i] == false) {
                dfs(i);
            }
        }

    }

    static void bfs(int start) {
        Queue<Integer> que = new LinkedList<Integer>();
        que.offer(start);
        visited[start] = true;

        while (!que.isEmpty()) {
            int node = que.poll();

            for (int i = 1; i <= n; i++) {
                if (graph[node][i] == 1 && visited[i] == false ) {
                    que.offer(i);
                    visited[i] = true;
                }
            } 

        }
    }
}